src/main/java/com/hierynomus/smbj/share/DiskShare.java

```diff
@@ -105,56 +81,63 @@ public Directory openDirectory(
         String path,
         EnumSet<AccessMask> accessMask,
         EnumSet<SMB2ShareAccess> shareAccess,
-        SMB2CreateDisposition createDisposition) throws TransportException, SMBApiException {
-        logger.info("OpenDirectory {},{},{},{},{}", path, accessMask, shareAccess, createDisposition);
-
-        SMB2FileId fileId = open(path, toLong(accessMask), EnumSet.of(FILE_ATTRIBUTE_DIRECTORY), shareAccess, createDisposition, EnumSet.of(SMB2CreateOptions.FILE_DIRECTORY_FILE));
-        return new Directory(fileId, treeConnect, path);
-
+        SMB2CreateDisposition createDisposition) throws SMBApiException {
+        return (Directory) open(path, toLong(accessMask), EnumSet.of(FILE_ATTRIBUTE_DIRECTORY), shareAccess, createDisposition, EnumSet.of(SMB2CreateOptions.FILE_DIRECTORY_FILE));
     }
 
     /**
      * Get a handle to a file
      */
-    public File openFile(String path, EnumSet<AccessMask> accessMask, SMB2CreateDisposition createDisposition) throws TransportException, SMBApiException {
-        logger.info("OpenFile {},{},{}", path, accessMask, createDisposition);
-
-        long accessMaskValue = toLong(accessMask);
-        SMB2FileId fileId = open(path, accessMaskValue, null, EnumSet.of(FILE_SHARE_READ), createDisposition, EnumSet.of(SMB2CreateOptions.FILE_NON_DIRECTORY_FILE));
-        // TODO
-        return new File(fileId, treeConnect, path, accessMaskValue);
+    public File openFile(String path, EnumSet<AccessMask> accessMask, SMB2CreateDisposition createDisposition) throws SMBApiException {
+        return (File) open(path, toLong(accessMask), EnumSet.of(FILE_ATTRIBUTE_DIRECTORY), EnumSet.noneOf(SMB2ShareAccess.class), createDisposition, EnumSet.of(SMB2CreateOptions.FILE_DIRECTORY_FILE));
```

Comment:
Are the passed options correct? They're the same as for the `openDirectory` call.